`find_package(iceoryx_posh)` fails

## Required information

**Operating system:**
Ubuntu 20.04 LTS

**Compiler version:**
E.g. GCC 9.3.0

**Observed result or behaviour:**
This is related to https://github.com/ros2/rmw_iceoryx/issues/61, a line `find_package(iceoryx_posh)` in a CMakeLists.txt from that repository fails with the error:

```
CMake Error at /usr/share/cmake-3.16/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findiceoryx_hoofs.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "iceoryx_hoofs", but CMake did not find one.

  Could not find a package configuration file provided by "iceoryx_hoofs"
  with any of the following names:

    iceoryx_hoofsConfig.cmake
    iceoryx_hoofs-config.cmake

  Add the installation prefix of "iceoryx_hoofs" to CMAKE_PREFIX_PATH or set
  "iceoryx_hoofs_DIR" to a directory containing one of the above files.  If
  "iceoryx_hoofs" provides a separate development package or SDK, be sure it
  has been installed.
```
I think the problem comes from  iceory_posh's package.xml only listing `iceoryx_hoofs` as a `build_depend`, and not a `depend`. Changing it to a `depend` tag fixes the issue.

**Expected result or behaviour:**
`find_package(iceoryx_posh REQUIRED)` to work.

**Conditions where it occurred / Performed steps:**
Building iceoryx and rms_iceoryx in a ROS2 workspace,
```
mkdir -p ~/iceoryx_ws/src
cd $_
git clone https://github.com/eclipse/iceoryx.git
git clone https://github.com/ros2/rmw_iceoryx.git
```

--- a/iceoryx_posh/package.xml
+++ b/iceoryx_posh/package.xml
@@ -13,7 +13,7 @@
     <buildtool_depend>cmake</buildtool_depend>
     <buildtool_depend>git</buildtool_depend>
 
-    <build_depend>iceoryx_hoofs</build_depend>
+    <depend>iceoryx_hoofs</depend>
 
     <doc_depend>doxygen</doc_depend>
 

